### ISSUE TITLE
Don't pin package dependencies in setup.py

### ISSUE THREAD

**Issue by fbergroth** [zalandoresearch/flair#420]
To be removed, once it is done: Please add the appropriate label to this ticket, e.g. feature or enhancement.

**Is your feature/enhancement request related to a problem? Please describe.**

It is not considered good practice to pin package dependencies in setup.py (see additional context).

For instance, I'm forced to downgrade certain packages by installing flair.

**Describe the solution you'd like**

Just list the abstract requirements in setup.py with less restrictive version bounds.

**Additional context**

See https://packaging.python.org/discussions/install-requires-vs-requirements/

**Comment by alanakbik** [comment]
Good point! Care to do a PR? :)

Which packages are you currently forced to downgrade?

**Comment by fbergroth** [comment]
> Care to do a PR? :)

Sure, but I'm not very familiar with flair.

I looked through the dependencies, and `typing` can be removed, since it's a backport for python < 3.5, but you have `python_requires='>=3.6'`.

For the rest, we could replace `==` by `>=`.

Regarding `pytorch-pretrained-bert==0.3.0`, will it work with `0.4.0`, or should that dependency stay pinned?

> Which packages are you currently forced to downgrade?

In my environment I had the latest versions, which caused gensim, tqdm to be downgraded.

**Comment by stefan-it** [comment]
I accidentally only updated `pytorch-pretrained-bert` to 0.4.0 in `requirements.txt` and not in `setup.py`, so you can definitely update it to *0.4.0* (I'm using that version in all of my experiments).

**Comment by fbergroth** [comment]
Btw, I recommend using `pip-compile` from [pip-tools](https://github.com/jazzband/pip-tools) to generate the requirements.txt.

If you want to keep the Pipfile, you probably want to commit Pipfile.lock as well, even though it's a bit redundant to have both Pipfile.lock and requirements.txt.
